drivers/hyperv/hyperv.go

```diff
@@ -186,20 +194,35 @@ func (d *Driver) PreCreateCheck() error {
 		return err
 	}
 
-	// Downloading boot2docker to cache should be done here to make sure
+	// Downloading boot2docker/windows-server to cache should be done here to make sure
 	// that a download failure will not leave a machine half created.
 	b2dutils := mcnutils.NewB2dUtils(d.StorePath)
-	err = b2dutils.UpdateISOCache(d.Boot2DockerURL)
+
+	if mcnutils.ConfigGuest.GetGuestOS() != "windows" {
+		err = b2dutils.UpdateISOCache(d.Boot2DockerURL)
+	} else {
+		err = b2dutils.UpdateVHDCache(d.WindowsVHDUrl)
+	}
+
 	return err
 }
 
 func (d *Driver) Create() error {
 	b2dutils := mcnutils.NewB2dUtils(d.StorePath)
-	if err := b2dutils.CopyIsoToMachineDir(d.Boot2DockerURL, d.MachineName); err != nil {
-		return err
+
+	if mcnutils.ConfigGuest.GetGuestOS() == "windows" {
```

Comment:
It'd be nice if the ConfigGuest type had a "IsWindows()" method that we reuse throughout this change. I know it's just a comparison against a string, but this type of comparison changes over time.